ui/components/app/gas-details-item/gas-details-item.js

```diff
@@ -14,10 +20,39 @@ import LoadingHeartBeat from '../../ui/loading-heartbeat';
 import GasTiming from '../gas-timing/gas-timing.component';
 import TransactionDetailItem from '../transaction-detail-item/transaction-detail-item.component';
 import UserPreferencedCurrencyDisplay from '../user-preferenced-currency-display';
+import { hexWEIToDecGWEI } from '../../../../shared/modules/conversion.utils';
 import GasDetailsItemTitle from './gas-details-item-title';
 
 const GasDetailsItem = ({ userAcknowledgedGasMissing = false }) => {
   const t = useI18nContext();
+  const draftTransaction = useSelector(getCurrentDraftTransaction);
+  const unapprovedTxs = useSelector(getUnapprovedTransactions);
+  let transactionData = {};
+  if (Object.keys(draftTransaction).length !== 0) {
+    const editingTransaction = unapprovedTxs[draftTransaction.id];
+    transactionData = {
+      txParams: {
+        gasPrice: draftTransaction.gas?.gasPrice,
+        gas: editingTransaction?.userEditedGasLimit
+          ? editingTransaction?.txParams?.gas
+          : draftTransaction.gas?.gasLimit,
+        maxFeePerGas: editingTransaction?.txParams?.maxFeePerGas
+          ? editingTransaction?.txParams?.maxFeePerGas
+          : draftTransaction.gas?.maxFeePerGas,
+        maxPriorityFeePerGas: editingTransaction?.txParams?.maxPriorityFeePerGas
+          ? editingTransaction?.txParams?.maxPriorityFeePerGas
+          : draftTransaction.gas?.maxPriorityFeePerGas,
+        value: draftTransaction.amount?.value,
+        type: draftTransaction.transactionType,
+      },
+      userFeeLevel: editingTransaction?.userFeeLevel,
+    };
+  }
```

Comment:
Logic above looks duplicated between legacy and EIP-1559 component, it will be useful to create a re-usable hook for this.